cdap-ui/app/features/adapters/services/adapter-detail-factory.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+angular.module(PKG.name + '.feature.adapters')
+  .factory('AdapterDetail', function(myWorkFlowApi, myMapreduceApi, myWorkersApi, GLOBALS) {
+
+    var publicObj = {
+      initialize: initialize
+    };
+
+    function initialize(app, $state) {
+      publicObj.programType = app.artifact.name === GLOBALS.etlBatch ? 'WORKFLOWS' : 'WORKERS';
+      publicObj.params = {
+        namespace: $state.params.namespace,
+        appId: app.name
+      };
+
+      publicObj.logsParams = {
+        namespace: $state.params.namespace,
+        appId: app.name,
+        max: 50
+      };
+
+      var config;
+      try {
+        config = JSON.parse(app.configuration);
+      } catch (e) {
+        console.log('ERROR cannot parse configuration');
```

Comment:
Same here. We should return in the catch block as subsequent statements become invalid beyond this point if there is a JSON parse error.